packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -174,8 +174,21 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateProps<T>)
       lastValue.current = inputValue;
     }
 
+    // Update the inputValue if the selected item's text changes from its last tracked value.
+    // This is to handle cases where a selectedKey is specified but the items aren't available (async loading) or the selected item's text value updates.
+    // Only reset if the user isn't currently within the field so we don't erroneously modify user input.
+    // If inputValue is controlled, it is the user's responsibility to update the inputValue when items change.
+    let selectedItemText = collection.getItem(selectedKey)?.textValue ?? '';
+    if (!isFocused && selectedKey != null && props.inputValue === undefined && selectedKey === lastSelectedKey.current) {
```

Comment:
Kept the `!isFocused` check since its possible that `items`/map of items provided to the ComboBox changes while the user is in the field. Figured we don't want to change the input value on the user, feels like a strange UX

Reply:
Makes sense. I guess it would reset on blur already